Feature/testing.v7

NWSchallock opened the pull request:
PTAL

NWSchallock committed:
Velocity control mode

bookslifestyle committed:
Fixed and Organized Buttons

NWSchallock committed:
Made intake speed variable based on drive speed

NWSchallock committed:
Made intake speed variable based on drive speed

NWSchallock committed:
rebase error fix

bookslifestyle committed:
Added Soft Limit Switch to Telescope

NWSchallock committed:
telescope tuning

bookslifestyle committed:
Set Up Position Control for the Winch

bookslifestyle committed:
Merge remote-tracking branch 'origin/feature/testing.v7' into feature/testing.v7

duane701 commented on src/main/cpp/subsystems/Winch.cpp:
```diff
@@ -7,12 +7,30 @@
 
 #include "subsystems/Winch.h"
 
+constexpr double kMaxVelocityError{4000-3000};
```
These numbers seem like they were copied from DooHickey. We need to do some measurements for the Winch.

duane701 commented on src/main/cpp/subsystems/Winch.cpp:
```diff
@@ -21,7 +39,8 @@ void Winch::Periodic()
 }
 
-void Winch::WinchHook(double percentoutput)
+void Winch::WinchHook(double position)
```
We should add a new interface. We probably want a simple test that doesn't involve the control system.

duane701 commented on src/main/cpp/subsystems/Winch.cpp:
```diff
@@ -7,12 +7,30 @@
+constexpr double kP{(.05*1023)/kMaxVelocityError};
+constexpr double kI{0.0};
+constexpr double kD{0.0};
+double kF{0.0};
```
`constexpr`

duane701 commented on src/main/include/RobotContainer.h:
```diff
@@ -81,7 +81,7 @@ class RobotContainer {
   Telescope mTelescope{"Telescope", mTelescopeComponents};
 
-  Winch::Components mWinchComponents{3,4};
+  Winch::Components mWinchComponents{5,6};
```
Use ID names.

duane701 commented on src/main/cpp/subsystems/Telescope.cpp:
```diff
@@ -6,24 +6,33 @@
+    mComponents.telescopeMotor.ConfigForwardSoftLimitEnable(false);
+    mComponents.telescopeMotor.ConfigReverseSoftLimitEnable(true);
     mComponents.telescopeMotor.ConfigForwardSoftLimitThreshold(1300);
+    mComponents.telescopeMotor.ConfigReverseSoftLimitThreshold(0);
```
Keep in mind that if the telescope is not down when the system reboots the lower threshold will be incorrect.

duane701 commented on src/main/cpp/commands/IntakeOn.cpp:
```diff
@@ -21,7 +26,9 @@ void IntakeOn::Initialize() {}
 {
-  mIntake.SetIntake(mSpeed);
+  double averageDrive = (mChassis.GetLeftVelocity() + mChassis.GetRightVelocity()) / 2;
+  double velocity = averageDrive * conversionFactor + mSpeed;
```
Technically this is speed not velocity.

bookslifestyle committed:
Made minor code changes for the winch

added constexpr, put speedcontrollergroup back in, and used id names
still need to figure out the number for velocity error

toxicoreos committed:
Shooter Tuning. 

Lots of tuning of the shooter. These are pretty good values for the short shot.

duane701 committed:
IntakeOn: Fix compiler warning

duane701 committed:
Remove commented code

duane701 committed:
Shooter: ControlMode to Implementation

ControlMode is only needed in the implementation.

duane701 committed:
Winch: ControlMode to implementation

ControlMode is only needed in the implementation

duane701 committed:
Shooter: SetPID is implementation

Move SetPID to the implementation.

No need for SetPID to be in the header.

duane701 committed:
Winch: SetPID is implementation

Move SetPID to implementation.

duane701 committed:
Shooter: constexpr

Lots of constexpr adjustments. Yay!

duane701 committed:
Shooter: Init mThresholdLoops

Oops! Initialize all the things.

duane701 committed:
WInch: WinchHookPercent

WinchHootPercent is a different from WinchHook. 

WinchHootPercent is for testing.

duane701 commented:
LGTM

duane701 reviewed (approved):
LGTM
